aip/client-libraries/4222.md

```diff
@@ -154,6 +154,41 @@ If none of the routing parameters matched their respective fields, the routing h
 Much like URL parameters, if there is more than one key-value pair to be sent, the `&`
 character is used as the separator.
 
+### `path_template` syntax
+
+As seen in the above examples, the `path_template` can use a variety of symbols
+that are interpreted by code generators during conversion to regular expressions
+or non-regular expression matcher implementations. The syntax for these symbols
+is as follows:
+
+- Segment delimiters:
+  `path_template`:
+  - `/`: The traditional delimiter for resource names and HTTP paths.
+  - `:`: Often, but not exclusively, used to separate a path segment from a
```

Comment:
`:` should only be used as the last delimeter as of now, do we want to specify that as well? Or something like `project:*/locations/**` is possible?

Reply:
I think we _want_ to be strict and we _have_ to be strict are two different things . If something like `project:*/locations/**` is valid than we can definitely support it, but if it's not a valid URL then I think we have to enforce it in path templates a well. I know `:` is generally allowed in URL, not sure if we have any internal guideline on the service side that the URLs have to follow certain patterns. 
Also do we treat `:` as a delimiter or a literal in this case? It may not matter for a regex based solution but it matters for a non-regex based solution.

Reply:
Ok so we decided to *not* support `:` as a delimiter and I've added more strict language on where wildcards can be used. Resolving this.